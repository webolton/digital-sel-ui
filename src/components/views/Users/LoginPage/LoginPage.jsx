import React from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import userActions from 'actions/userActions';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';
import { Field, reduxForm } from 'redux-form';
import GridContainer from 'components/Grid/GridContainer';
import GridItem from 'components/Grid/GridItem';
import Button from 'components/CustomButtons/Button';
import Card from 'components/Card';
import CardBody from 'components/Card/CardBody';
import CardHeader from 'components/Card/CardHeader';
import CardFooter from 'components/Card/CardFooter';
import TextField from '@material-ui/core/TextField';
import loginPageStyle from 'assets/javascripts/views/users/loginPageStyle';

const validate = (values) => {
  const errors = {};
  const requiredFields = [
    'email',
    'password',
  ];
  requiredFields.forEach((field) => {
    if (!values[field]) {
      errors[field] = 'Required';
    }
  });
  if (
    values.email
    && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)
  ) {
    errors.email = 'Invalid email address';
  }
  return errors;
};

const renderTextField = ({
  label,
  input,
  propClasses,
  meta: { touched, invalid, error },
  ...custom
}) => (
  <TextField
    label={label}
    placeholder={label}
    error={touched && invalid}
    helperText={touched && error}
    InputProps={{
      classes: {
        root: propClasses.input,
        underline: propClasses.underline,
        error: propClasses.underlineError,
        disabled: propClasses.disabled,
      },
    }}
    InputLabelProps={{
      classes: {
        root: propClasses.labelRoot,
        error: propClasses.labelRootError,
      },
    }}
    FormHelperTextProps={{
      classes: {
        error: propClasses.customHelperTextError,
        success: propClasses.helperTextSuccess,
      },
    }}
    {...input}
    {...custom}
  />
);

class LoginPage extends React.Component {
  constructor(props) {
    super(props);
    this.submitLogin = this.submitLogin.bind(this);
  }

  submitLogin = (values) => {
    const { loginUser } = this.props;
    loginUser(values.email, values.password);
  }

  render() {
    const {
      classes,
      alert,
      handleSubmit,
      pristine,
      submitting,
    } = this.props;

    return (
      <div>
        <div
          className={classes.pageHeader}
          style={{
            backgroundSize: 'cover',
            backgroundPosition: 'top center',
          }}
        >
          <div className={classes.container}>
            <GridContainer justify="center">
              <GridItem xs={12} sm={12} md={4}>
                <Card>
                  <form
                    className={classes.form}
                    onSubmit={handleSubmit(val => this.submitLogin(val))}
                  >
                    <CardHeader color="primary" className={classes.cardHeader}>
                      <h4>Login</h4>
                    </CardHeader>
                    <CardBody>
                      {alert.message
                        && <p className={classes.unauthorized}>Incorrect email or password.</p>
                      }
                      <Field
                        name="email"
                        label="Email"
                        id="email"
                        component={renderTextField}
                        fullWidth
                        propClasses={classes}
                        classes={{
                          root: classes.formControl,
                        }}
                      />
                      <Field
                        name="password"
                        label="Password"
                        id="password"
                        type="password"
                        component={renderTextField}
                        fullWidth
                        propClasses={classes}
                        classes={{
                          root: classes.formControl,
                        }}
                      />
                    </CardBody>
                    <CardFooter className={classes.cardFooter}>
                      <Button
                        simple
                        color="success"
                        size="lg"
                        type="submit"
                        disabled={pristine || submitting}
                      >
                        submit
                      </Button>
                    </CardFooter>
                  </form>
                </Card>
              </GridItem>
            </GridContainer>
          </div>
        </div>
      </div>
    );
  }
}

LoginPage.defaultProps = {
  alert: {},
  pristine: false,
  submitting: false,
};

LoginPage.propTypes = {
  classes: PropTypes.object.isRequired,
  loginUser: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool,
  submitting: PropTypes.bool,
  alert: PropTypes.object,
};

export default compose(
  connect(
    state => ({ // eslint-disable-line no-unused-vars
    }),
    {
      loginUser: userActions.login,
    },
  ),
  reduxForm({
    form: 'LoginPage',
    validate,
    enableReinitialize: true,
  }),
  withStyles(loginPageStyle),
)(LoginPage);
