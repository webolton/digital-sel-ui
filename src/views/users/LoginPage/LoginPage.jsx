import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { userActions } from 'actions';
import withStyles from '@material-ui/core/styles/withStyles';
import InputAdornment from '@material-ui/core/InputAdornment';
import Icon from '@material-ui/core/Icon';
import Email from '@material-ui/icons/Email';
import People from '@material-ui/icons/People';
import Header from 'components/Header/Header';
import HeaderLinks from 'components/Header/HeaderLinks';
import Footer from 'components/Footer/Footer';
import GridContainer from 'components/Grid/GridContainer.jsx';
import GridItem from 'components/Grid/GridItem.jsx';
import Button from 'components/CustomButtons/Button.jsx';
import Card from 'components/Card/Card';
import CardBody from 'components/Card/CardBody.jsx';
import CardHeader from 'components/Card/CardHeader.jsx';
import CardFooter from 'components/Card/CardFooter.jsx';
import CustomInput from 'components/CustomInput/CustomInput.jsx';
import loginPageStyle from 'assets/javascripts/views/users/loginPageStyle';

class LoginPage extends React.Component {
  constructor(props) {
    super(props);

    // reset login status
    // this.props.dispatch(userActions.logout());

    this.state = {
      email: '',
      password: '',
      submitted: false,
      cardAnimaton: 'cardHidden',
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  }

  handleSubmit(e) {
    e.preventDefault();

    this.setState({ submitted: true });
    const { email, password } = this.state;
    const { dispatch } = this.props;
    if (email && password) {
      dispatch(userActions.login(email, password));
    }
  }

  componentDidMount() {
    // we add a hidden class to the card and after 700 ms we delete it and the transition appears
    setTimeout(
      function() {
        this.setState({ cardAnimaton: "" });
      }.bind(this),
      50
    );
  }

  render() {
    const { loggingIn, classes, ...rest } = this.props;
    const { email, password, submitted } = this.state;
    return (
      <div>
        <div
          className={classes.pageHeader}
          style={{
            backgroundSize: "cover",
            backgroundPosition: "top center"
          }}
        >
          <div className={classes.container}>
            <GridContainer justify="center">
              <GridItem xs={12} sm={12} md={4}>
                <Card className={classes[this.state.cardAnimaton]}>
                  <form className={classes.form}>
                    <CardHeader color="primary" className={classes.cardHeader}>
                      <h4>Login</h4>
                    </CardHeader>
                    <p className={classes.divider}>Or Be Classical</p>
                    <CardBody>
                      <CustomInput
                        labelText="First Name..."
                        id="first"
                        formControlProps={{
                          fullWidth: true
                        }}
                        inputProps={{
                          type: "text",
                          endAdornment: (
                            <InputAdornment position="end">
                              <People className={classes.inputIconsColor} />
                            </InputAdornment>
                          )
                        }}
                      />
                      <CustomInput
                        labelText="Email..."
                        id="email"
                        formControlProps={{
                          fullWidth: true
                        }}
                        inputProps={{
                          type: "email",
                          endAdornment: (
                            <InputAdornment position="end">
                              <Email className={classes.inputIconsColor} />
                            </InputAdornment>
                          )
                        }}
                      />
                      <CustomInput
                        labelText="Password"
                        id="pass"
                        formControlProps={{
                          fullWidth: true
                        }}
                        inputProps={{
                          type: "password",
                          endAdornment: (
                            <InputAdornment position="end">
                              <Icon className={classes.inputIconsColor}>
                                lock_outline
                              </Icon>
                            </InputAdornment>
                          )
                        }}
                      />
                    </CardBody>
                    <CardFooter className={classes.cardFooter}>
                      <Button simple color="primary" size="lg">
                        Get started
                      </Button>
                    </CardFooter>
                  </form>
                </Card>
              </GridItem>
            </GridContainer>
          </div>
          <Footer whiteFont />
        </div>




        <form name="form" onSubmit={this.handleSubmit}>
          <div className={'form-group' + (submitted && !email ? ' has-error' : '')}>
            <label htmlFor="email">Email!!</label>
              <input type="text" className="form-control" name="email" value={email} onChange={this.handleChange} />
                {submitted && !email &&
                  <div className="help-block">Email is required</div>
                }
          </div>
          <div className={'form-group' + (submitted && !password ? ' has-error' : '')}>
            <label htmlFor="password">Password</label>
            <input type="password" className="form-control" name="password" value={password} onChange={this.handleChange} />
              {submitted && !password &&
                <div className="help-block">Password is required</div>
              }
          </div>
          <div className="form-group">
            <button className="btn btn-primary">Login</button>
              {/* for a spinner */}
              { loggingIn }
          </div>
        </form>
      </div>
    );
  }
}

function mapStateToProps(state) {
  const { loggingIn } = state.authentication;
  return {
    loggingIn
  };
}

export default connect(mapStateToProps)(withStyles(loginPageStyle)(LoginPage));
