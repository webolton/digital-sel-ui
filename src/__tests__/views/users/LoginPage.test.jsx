import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import LoginPage from 'views/users/LoginPage';

const initialState = { authentication: '' };
const mockStore = configureStore();
const store = mockStore(initialState);

describe('<LoginPage />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<LoginPage store={store} />);
      const component = wrapper.dive();
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
