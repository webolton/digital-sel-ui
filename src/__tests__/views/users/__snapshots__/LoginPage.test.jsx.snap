// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> render() renders the component 1`] = `
<LoginPage
  alert={Object {}}
  classes={
    Object {
      "cardFooter": "LoginPage-cardFooter-8",
      "cardHeader": "LoginPage-cardHeader-5",
      "cardHidden": "LoginPage-cardHidden-2",
      "container": "LoginPage-container-1",
      "divider": "LoginPage-divider-7",
      "form": "LoginPage-form-4",
      "inputIconsColor": "LoginPage-inputIconsColor-10",
      "pageHeader": "LoginPage-pageHeader-3",
      "socialIcons": "LoginPage-socialIcons-6",
      "socialLine": "LoginPage-socialLine-9",
      "unauthorized": "LoginPage-unauthorized-11",
    }
  }
  dispatch={[Function]}
  email=""
  loggingIn={false}
  password=""
  rest={Object {}}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  submitted={false}
/>
`;
